stdlib/public/core/KeyPath.swift

```diff
@@ -3703,3 +3703,138 @@ internal func _instantiateKeyPathBuffer(
+    _: Base.Type,
+    _ leaf: Leaf.Type
+) -> String {
+    let getter: ComputedAccessorsPtr.Getter<Base, Leaf> = pointer.getter()
+    let pointer = unsafeBitCast(getter, to: UnsafeRawPointer.self)
+    if let cString = keypath_dladdr(UnsafeRawPointer(pointer)),
+        let demangled = demangle(name: cString) {
+        defer {
+            demangled.deallocate()
+        }
+        return String(cString: demangled)
+    } else {
+        return "<computed \(pointer) (\(leaf))>"
+    }
+}
+
+@available(SwiftStdLib 5.7, *)
+extension AnyKeyPath: CustomDebugStringConvertible {
+        
+    @available(SwiftStdLib 5.7, *)
+    public var debugDescription: String {
+        var description = "\\\(String(describing: Self.rootType))"
+        return withBuffer {
+            var buffer = $0
+            if buffer.data.isEmpty {
+                _internalInvariantFailure("key path has no components")
+            }
+            var valueType: Any.Type = Self.rootType
+            while true {
+                let (rawComponent, optNextType) = buffer.next()
+                let hasEnded = optNextType == nil
+                let nextType = optNextType ?? Self.valueType
+                func name(for offset: Int) -> String {
+                    #if SWIFT_ENABLE_REFLECTION
+                    let count = _getRecursiveChildCount(valueType)
+                    for i in 0..<count {
+                        if _getChildOffset(
+                            valueType,
+                            index: i
+                        ) == offset {
+                            var field = _FieldReflectionMetadata()
+                            _ = _getChildMetadata(
+                                valueType,
+                                index: i,
+                                fieldMetadata: &field
+                            )
+                            defer {
+                                field.freeFunc?(field.name)
+                            }
+                            return String(cString: field.name)
+                        }
+                    }
+                    #endif
+                    return "<offset \(offset) (\(nextType))>"
+                }
+                func name(
+                    for pointer: ComputedAccessorsPtr,
+                    base: Any.Type,
+                    leaf: Any.Type
+                ) -> String {
+                    func project<Base>(base: Base.Type) -> String {
+                        func project2<Leaf>(leaf: Leaf.Type) -> String {
+                            dynamicLibraryAddress(
+                                of: pointer,
+                                base,
+                                leaf
+                            )
+                        }
+                        return _openExistential(nextType, do: project2)
+                    }
+                    return _openExistential(base, do: project)
+                }
+                func addChain() {
```

Comment:
I think you can replace this with a single `description.append(".")` at the beginning of the loop.

Reply:
My first version of this function was implemented like that, but unfortunately I don't think it works. Optionals require some special handling, otherwise we end up with stuff like `\Theme.background.?alpha`.

Reply:
then I think you can it at the end of the loop and do a `description.removeLast()` once you're outside it.

Reply:
It wasn't quite that simple, but the suggestion gave me an idea for a different impl which I think addresses this.